tests/tests.wake

```diff
@@ -77,6 +87,77 @@ export def runTests (cmdline: List String): Result String Error =
 
     Pass "All {len tests | str} tests completed successfully."
 
+
+export def runUnitTests _: Result Unit Error =
+    def testDirectory = "{here}/wake-unit"
+    require Pass testScript =
+      source "{testDirectory}/unit-test.sh"
+
+    def testName =
+        testDirectory
+        | replace `^${quote here}/` ""
+
+    def readFile list = match list
+        Some file =
+            require Pass content = read file
+            Pass (Some content)
+        None =
+            Pass None
+
+    require Pass expectedStdout =
+        source "{testDirectory}/stdout"
+        | getPass
+        | readFile
+
+    require Pass expectedStderr =
+        source "{testDirectory}/stderr"
+        | getPass
+        | readFile
+
+    require Pass visibleFiles =
+        sources testDirectory `.*`
+
+    def inTestDir path =
+        relative testDirectory path
+
+    require Pass (Pair wakeBin wakeVisible) =
+        wakeToTest Unit
+
+    require Pass wakeUnitBin =
+        wakeUnitToTest Unit
+
+    def testJob =
+        makeExecPlan ("./{testScript.getPathName.inTestDir}", wakeUnitBin.inTestDir, Nil) (visibleFiles ++ wakeVisible)
+        | setPlanDirectory testDirectory
+        | setPlanLabel "testing: {testName}"
+        | setPlanLocalOnly True # On OS/X you cannot mount fuse within fuse
```

Comment:
`| setPlanLocalOnly True`
If this is a MacOS-only limitation, we could have an `if sysname == "Darwin"`.
If we have unconditional `setPlanLocalOnly` it causes the list of visible files to be ignored, which makes a few of the `require` statements above redundant

Reply:
I think I can remove that for this case. I cargo culted it from below.